test/retry_test.go

```diff
@@ -137,9 +137,7 @@ func (s) TestRetryMaxAttemptsUnary(t *testing.T) {
 		  }
 		}]}`, tc.serviceMaxAttempts)),
 		}
-		if tc.clientMaxAttempts >= 0 {
-			clientOpts = append(clientOpts, grpc.WithMaxRetryAttempts(tc.clientMaxAttempts))
-		}
+		clientOpts = append(clientOpts, grpc.WithMaxCallAttempts(tc.clientMaxAttempts))
```

Comment:
Nit, optional: don't use `append`.  Could also eliminate `clientOpts` in theory, but it's possible it aids readability, too.

Reply:
Ah yes. The `append` is superfluous now that the condition is removed.